hail/python/hail/linalg/blockmatrix.py

```diff
@@ -1427,6 +1459,47 @@ def __matmul__(self, b):
 
         return BlockMatrix(BlockMatrixDot(self._bmir, b._bmir))
 
+    @typecheck_method(b=oneof(np.ndarray, block_matrix_type), split_on_inner=int, path_prefix=str)
+    def tree_matmul(self, b, split_on_inner, path_prefix):
```

Comment:
I think we should make either the last or the last two parameters keyword-only arguments, so: `(self, b, *, split_on_inner, path_prefix)`.

Reply:
I'd be willing to make `path_prefix` a keyword only argument, but can you explain why you'd prefer that? I think `split_on_inner` I'll leave the way it is, since you always have to specify it. `path_prefix` with a default value of `None` as your above comment suggests makes the keyword only argument sound better.